client/core/core.go

```diff
@@ -1256,6 +1266,55 @@ func (c *Core) Login(pw []byte) (*LoginResult, error) {
 	return result, nil
 }
 
+func (c *Core) reinstateDexAccount(dexStat *DEXBrief, crypter encrypt.Crypter) (*LoginResult, error) {
+
+	dc := c.conns[dexStat.Host]
+
+	// Prepare and sign the registration payload.
+	// The account ID is generated from the public key.
+	dexReg := &msgjson.Reinstate{
+		DEXPubKey:    dc.acct.dexPubKey.Serialize(),
+		ClientPubKey: dc.acct.privKey.PubKey().SerializeCompressed(),
+		AccountProof: dc.acct.accountProof,
+		CoinID:       dc.acct.feeCoin,
+		Time:         encode.UnixMilliU(time.Now()),
+	}
+	regRes := new(msgjson.RegisterResult)
+	err := dc.signAndRequest(dexReg, msgjson.ReinstateRoute, regRes)
+	if err != nil {
+		log.Errorf("error reinstating account on dex: %v", err)
+		return nil, err
+	}
+
+	regFeeAssetID, _ := dex.BipSymbolID(regFeeAssetSymbol)
+	dcrWallet, err := c.connectedWallet(regFeeAssetID)
+	if err != nil {
+		return nil, fmt.Errorf("cannot connect to %s wallet to pay fee: %v", regFeeAssetSymbol, err)
```

Comment:
I put it there so that the wallet would stop locking in a reinstate case. On second thought it is probably most secure to make the user explicitly unlock the wallet. I will remove it.

Reply:
I just meant the wording of the message. The fee is already paid, I think, so the purpose is not `to pay fee`. Sorry I wasn't more explicit.

Reply:
No problem.